### PR TITLE
Add documentation for auth body extractor

### PR DESCRIPTION
**Pull request by shadowhand**
Refs sparkphp/auth#10

### CODE REVIEW

File: docs/auth.md [hunk 1]
```diff
@@ -43,20 +43,22 @@ This method of extraction is represented by an implementation of [`Credentials\E
 
 These credentials extractor implementations are bundled with this library:
 
-* [`JsonExtractor`](https://github.com/sparkphp/auth/blob/master/src/Credentials/JsonExtractor.php) extracts the credentials from top-level properties of a JSON request body. Its constructor takes the names of the properties containing the user identifier and password.
+* [`Body`](https://github.com/sparkphp/auth/blob/master/src/Credentials/BodyExtractor.php) extracts the credentials from top-level properties of the parsed request body. Its constructor takes the names of the properties containing the user identifier and password.
```
Comment (elazar) [inline docs/auth.md]: Think this should reference the entire class name to be consistent with class name references in the rest of the document, as it is in the changes below this one.

File: docs/auth.md [hunk 2]
```diff
@@ -43,20 +43,22 @@ This method of extraction is represented by an implementation of [`Credentials\E
...
+use Spark\Auth\Credentials\BodyExtractor;
 
 $injector->alias(
     ExtractorInterface::class,
-    JsonExtractor::class
+    BodyExtractor::class
 );
 ```
 
+**Note**: When using `BodyExtractor` it is expected that any [`*ContentHandler` middleware](http://spark.readthedocs.org/en/latest/#middleware) will be placed *before* `AuthHandler` to ensure that the request body has been parsed before authentication is attempted.
```
Comment (elazar) [inline docs/auth.md]: Extra asterisk before `ContentHandler` here.

File: docs/auth.md (hunk 1 above)
Comment (shadowhand) [inline docs/auth.md]: Whoops. Not sure how I missed that.

File: docs/auth.md (hunk 2 above)
Comment (shadowhand) [inline docs/auth.md]: Because there are several... `JsonContentHandler`, `FormContentHandler`, etc.

Comment (shadowhand) [inline docs/auth.md]: Open to suggestions how to make that more clear.

File: docs/auth.md [hunk 3]
```diff
@@ -43,20 +43,22 @@ This method of extraction is represented by an implementation of [`Credentials\E
 
 These credentials extractor implementations are bundled with this library:
 
-* [`JsonExtractor`](https://github.com/sparkphp/auth/blob/master/src/Credentials/JsonExtractor.php) extracts the credentials from top-level properties of a JSON request body. Its constructor takes the names of the properties containing the user identifier and password.
+* [`BodyExtractor`](https://github.com/sparkphp/auth/blob/master/src/Credentials/BodyExtractor.php) extracts the credentials from top-level properties of the parsed request body. Its constructor takes the names of the properties containing the user identifier and password.
```
Comment (shadowhand) [inline docs/auth.md]: @elazar fixed.

File: docs/auth.md (hunk 2 above)
Comment (elazar) [inline docs/auth.md]: Ah, I see. Fair enough.

**Comment by elazar** [comment]
:+1:

**Pull request merged**